### DISCUSSION TITLE
Connecting to SDongleA-05 Version V200R022C10SPC108

### DISCUSSION THREAD

**Discussion by jnikodemus** [wlcrs/huawei_solar#651] [Setting up the integration] [answered]
Hi,

I'm trying to get the integration running, but it wont connect to the inverter... I can open the connection using telnet on port 502, but cant get the integration running (tried slaveId 0 & 1). As I'm new to this, I don't know if the Version V200R022C10SPC108 is implicit covered by the integration. Can someone enlighten me?

**Comment by ManfredGiesinger** [comment]
My integration is running with SDongleA-05 Version V200R022C10SPC108 and SUN2000-10KTL-M1 V100R001C00SPC153 since March 16th. 

Documentation mention: 
This integration supports inverters running V200R001C00 firmware.
If you are running an older V100R001C00 firmware, then please consult the "Upgrade guide" included with V200R001C00 firmware versions to check if your inverter is eligibile for an upgrade.

That was the main reason I just have upgraded at least SDongleA to the V200, my SUN2000 only support 100. 

Usually the integration works fine, but once a day I lose the connection to inverter, even it came back after some time. 

I must use modbus-proxy, as I have an additional device (Smartfox) which access the inverter.
Through the installation process I must disable the additional device.
So it works, even no improvement for my issue, where I will look a little bit closer the next few days.

**Reply by Roving-Ronin** [reply]
@ManfredGiesinger  just to clarify, re " This integration supports inverters running V200R001C00 firmware". The V200R001C00 firmware is the firmware for L1 / single phase inverters.  The M1 / 3 phase inverters use V100R001C00SPCxxx firmware.
Confusingly the sDongleA-05 can operate V100R001C00SPCxxx firmware, but moved to V200R022C10SPCxxx last year, with the newer version also being required to support newer products like EMMA / SmartGuard / LUNA S1 that are coming to market.

@jnikodemus please refer to the Wiki that lists the firmware versions that other people have used and found works. This list has been updated to break it up into different sections for the different inverter models (sorted by size of the inverter, and then version number of the firmware upon that model) as well as adding tracking firmware being used for LUNA batteries also.

https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions

**Reply by ManfredGiesinger** [reply]
Thank you for clarifying the situation with the different kind off Huawei Inverters and sDongle. 
(I only mention what was writen in hacs repository and will add my final versions in the next few days.)
Unfortunately, I was too stupid to find the page with the confirmed-firmware-version, before you provide the link. 
I like the integration but had some stability issues. Anyway, today I have looked once more in the Wiki and have changed now the connection from sDongle (Option B) to the Inverter directly (Option A). Looking forward if this was the right solution or if I have to investigate more.

**Reply by Roving-Ronin** [reply]
@ManfredGiesinger  
IMHO no way I'd use the inbuilt WiFi AP of the M1, as the SDongle is so much easier and works. With 2 x 6kW M1's I've been running V100R001C00SPC160 (and prior to that SPC155 for months) with the sDongle on V200R022C10SPC108 (and last 2 month SPC110) and its extremely stable with no outages / dropouts.
Connection is via WiFi (for some reason more stable than the ethernet of the dongle)  to UniFi nanoHD AP, with a IoT SSID set to 2.4Ghz only and allowing for 'down to' 1Mbps connections. Also remember to disable the nightly 'WiFi optimisations' that UniFi does for that AP/SSID.

**Reply by ManfredGiesinger** [reply]
@Roving-Ronin 
Very interesting that the WiFi is more stable as the LAN connection!
I will keep this in mind, as my SDongle today is connected through LAN and I agree that I would prefer the connection through SDongle too.

**Comment by jnikodemus** [comment, marked as answer]
> https://github.com/wlcrs/huawei_solar/wiki/Confirmed-firmware-versions

I've checked that list, but my inverter doesn't seem to be tested till now...
SUN2000-25K-MB0 [V200R023C10SPC111]
SDongleA-05 [V200R022C10SPC108]

Guess I'll have to wait.

Edit:
Well... I was just testing SlaveIDs and tried "2". Now it‘s working.
